### PR TITLE
feat(icon-button): add size modifiers

### LINKED ISSUE #1884: eBay-icon-button: add small 32px option
We've added a smaller icon button style for icon buttons. 
[Storybook component](https://opensource.ebay.com/ebayui-core/?path=/docs/buttons-ebay-icon-button--standard)
[Playbook](https://playbook.corp.ebay.com/PlaybookApp#/product-guidelines/patterns/buttons)
[Figma](https://www.figma.com/file/zEBdEhbonrBOGzZ0fXzWvM/eBay-Design-System?node-id=65935%3A91852&viewport=588%2C59%2C0.11)

image

### PR DESCRIPTION
**Pull request by LuLaValva**
- Fixes #1884 

- [x] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description

Add size modifiers to icon button and icon link

## Notes

- SUBJECTIVE CHANGE: Icons typically filled most of the button when a small modifier was applied, so I reduced `max-width` from `100%` to `75%`. Other values may be better (i.e. if we want to be mathematically precise and fill the circle with square icons it should be `70.71%`)

## Screenshots

image

## Checklist

- [ ] I verify the build is in a non-broken state
- [ ] I verify all changes are within scope of the linked issue

- [x] I regenerated all CSS files under dist folder
- [ ] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [x] I tested the UI in dark mode and RTL mode
- [ ] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

File: src/less/icon-button/icon-button.less [hunk 1]
```diff
@@ -19,15 +19,14 @@ a.icon-link {
...
     margin: 0;
-    min-width: 40px;
     padding: 0;
     vertical-align: text-bottom;
     width: 40px;
 
     > svg {
         .fill-token(icon-button-icon-foreground-color, color-foreground-primary);
 
-        max-width: 100%; // helps when width & height HTML attrs not set
```
Comment (agliga) [inline src/less/icon-button/icon-button.less:30]: Why was this changed?

**Review by agliga — APPROVED** [review]
LGTM, approving, but I have one question.

**Review by ArtBlue — APPROVED** [review]

File: src/less/icon-button/icon-button.less (hunk 1 above)
Comment (LuLaValva) [reply src/less/icon-button/icon-button.less:30]: I discussed this change in the notes for the PR, would like to hear your thoughts on it:

> Icons typically filled most of the button when a small modifier was applied, so I reduced `max-width` from `100%` to `75%`. Other values may be better (i.e. if we want to be mathematically precise and fill the circle with square icons it should be `70.71%`)

Comment (agliga) [reply src/less/icon-button/icon-button.less:30]: Lets put it in. I think it's fine.
